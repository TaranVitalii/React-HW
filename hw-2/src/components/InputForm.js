import React from "react";
import UserIdInput from "./UserIdInput";
import ButtonSubmit from "./ButtonSubmit";

export default class InputForm extends React.Component {
  state = {};
  onChange = dataFromForm => {
    this.setState({ dataFromForm }, () => console.log("onChange", this.state));
  };

  render() {
    return (
      <form className="user-form">
        <UserIdInput onChange={this.onChange} />
        <ButtonSubmit
          onSubmit={this.props.onclick}
          data={this.state.dataFromForm}
        />
      </form>
    );
  }
}
