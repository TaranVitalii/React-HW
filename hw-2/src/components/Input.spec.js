import React from "react";
import InputForm from "./Input";
import { mount, shallow } from "enzyme";

describe("<InputForm/>", () => {
  it("should render a InputForm", () => {
    const wrapper = shallow(<InputForm />);
    expect(wrapper).toMatchSnapshot();
  });

  //   it("should open a code editor", () => {
  //     const wrapper = mount(<InputForm code={code} />);
  //     expect(wrapper.find(".ReactCodeMirror")).toHaveLength(0);
  //     wrapper.find(".Button").simulate("click");
  //     expect(wrapper.find(".ReactCodeMirror")).toHaveLength(1);
  //   });
});
