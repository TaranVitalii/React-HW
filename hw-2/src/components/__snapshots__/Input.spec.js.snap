// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputForm/> should render a InputForm 1`] = `ShallowWrapper {}`;
